packages/fxa-content-server/app/scripts/views/pair/index.js

```diff
@@ -21,22 +22,27 @@ class PairIndexView extends FormView {
     const uap = this.getUserAgent();
     const isFirefoxDesktop = uap.isFirefoxDesktop();
 
-    if (! isFirefoxDesktop) {
-      // other browsers show an unsupported screen
+    if (! isFirefoxDesktop || ! this.broker.hasCapability('supportsPairing')) {
+      // other browsers show an unsupported screen or if no capability to pair
       return this.replaceCurrentPage('pair/unsupported');
     }
 
+    const account = this.broker.get('browserSignedInAccount');
     // If we reach this point that means we are in Firefox Desktop
-    if (! this.broker.get('browserSignedInAccount')) {
-      // if we are not logged into Sync then we offer to sign in
+    if (! account) {
+      // if we are not logged into Sync then we offer to connect
       return this.replaceCurrentPage('connect_another_device');
     }
 
-    if (! this.broker.hasCapability('supportsPairing')) {
-      return this.replaceCurrentPage('pair/unsupported');
+    if (! account.verified || ! account.sessionToken) {
+      // if account is not verified or missing sessionToken then offer to sign in or confirm
```

Comment:
Is it possible for the account to be missing a sessionToken? Does that happen in PB mode?

Reply:
not in PB mode, but it happened to somebody in user research :(

Reply:
> not in PB mode, but it happened to somebody in user research :(

Amazing... :( seems about right.